modules/cnv_upload_virtctl.adoc

```diff
@@ -1,51 +1,50 @@
 [[upload-vmdisk-virtctl]]
 === Uploading a local disk image to a new PVC
 
-You can use *virtctl image-upload* to upload a virtual machine disk image from 
-a client machine to your {product-title} cluster. This creates a PVC that can be 
+You can use `virtctl image-upload` to upload a virtual machine disk image from
+a client machine to your {product-title} cluster. This creates a PVC that can be
 associated with a virtual machine after the upload has completed.
 
 .Prerequisites
 
-* A virtual machine disk image, in RAW or QCOW2 format. It can be compressed 
+* A virtual machine disk image, in RAW or QCOW2 format. It can be compressed
 using *xz* or *gzip*.
 * *kubevirt-virtctl* must be installed on the client machine.
+* The client machine must be xref:../install_config/router/default_haproxy_router.html#overview[configured]
+to trust the OpenShift router's certificate.
 
 .Procedure
 
 . Identify the following items:
 * File location of the VM disk image you want to upload
 * Name and size desired for the resulting PVC
- 
-. Expose the *cdi-uploadproxy* service so that you can upload data to your cluster:
+
+. Remove the existing passthrough route:
++
+----
+$ oc delete route -n cdi cdi-uploadproxy-route
+----
+
+. Create a secured route using re-encryption termination:
 +
 ----
-cat <<EOF | oc apply -f -
-apiVersion: v1
-kind: Route
-metadata:
-  name: cdi-uploadproxy
-  namespace: kube-system 
-spec:
-  to:
-    kind: Service
-    name: cdi-uploadproxy 
-  tls:
-    termination: passthrough
-EOF
-----
-
-. Use the `virtctl image-upload` command to upload your VM image, 
+$ oc get secret -n cdi cdi-upload-proxy-ca-key -o=jsonpath="{.data['tls\.crt']}" | base64 -d > ca.pem
+
+$ oc create route reencrypt -n cdi --service=cdi-uploadproxy --dest-ca-cert=ca.pem
+----
+
+. Use the `virtctl image-upload` command to upload your VM image,
 making sure to include your chosen parameters. For example:
 +
 ----
-$ virtctl image-upload --uploadproxy-url=https://$(oc get route cdi-uploadproxy -o=jsonpath='{.status.ingress[0].host}') --pvc-name=upload-fedora-pvc --pvc-size=10Gi --image-path=/images/fedora28.qcow2
+$ virtctl image-upload --uploadproxy-url=https://$(oc get route cdi-uploadproxy-route -n cdi -o=jsonpath='{.status.ingress[0].host}') --pvc-name=upload-fedora-pvc --pvc-size=10Gi --image-path=/images/fedora28.qcow2
```

Comment:
@ousleyp s/cdi-uploadproxy-route/cdi-uploadproxy

Reply:
Let's keep the route name the same as kubevirt-ansible originally created it.  What we are trying to accomplish is replace the original passthrough route with a reeincrypyt route while keeping the rest of the steps unchanged.